### PR TITLE
New rule 'no-unguarded-host-mutation' 

### PR DESCRIPTION
**Pull request by Shinoni**
Error programmatic CSS during hydration (host mutation)

### CODE REVIEW

**Review by lpomerleau — COMMENTED** [review]
Thanks! I have a question out to the LWC team to ask if host mutations are allowed in CSRed components, or if this rule is only for hydrating SSRed components.

**Comment by lpomerleau** [comment]
> It's not SSR-specific. It's a bad idea even in pure-CSR

So this rule shouldn't be scoped to just "hydration". LWC team recommends to never mutate the host in the `connectedCallback`.

**Comment by Shinoni** [comment]
> > It's not SSR-specific. It's a bad idea even in pure-CSR
> 
> So this rule shouldn't be scoped to just "hydration". LWC team recommends to never mutate the host in the `connectedCallback`.

Good to know. I'll remove any reference to 'hydration' and will make it generic.

**Review by lpomerleau — APPROVED** [review]
Thanks! Which preset(s) will this rule be added to?
https://github.com/salesforce/eslint-config-lwc?tab=readme-ov-file#configurations

**Comment by Shinoni** [comment]
> Thanks! Which preset(s) will this rule be added to? https://github.com/salesforce/eslint-config-lwc?tab=readme-ov-file#configurations

It's in the "LWC" category -> recommended.

File: lib/rules/no-host-mutation-in-connected-callback.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const { docUrl } = require('../util/doc-url');
+
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {
+            description: 'Disallow mutations to the host element in connectedCallback',
+            category: 'LWC',
+            recommended: true,
+            url: docUrl('docs/rules/no-host-mutation-in-connected-callback-during-hydration'),
```
Comment (lpomerleau) [inline lib/rules/no-host-mutation-in-connected-callback.js:18]: says hydration

**Review by lpomerleau — APPROVED** [review]
There's just 1 instance of "hydration" in the rule to remove.

File: lib/rules/no-host-mutation-in-connected-callback.js (hunk 1 above)
Comment (Shinoni) [reply lib/rules/no-host-mutation-in-connected-callback.js:18]: Good catch!

**Pull request merged**